GitTfs/Core/GitTfsRemote.cs

```diff
@@ -325,9 +325,12 @@ public IFetchResult FetchWithMerge(int mergeChangesetId, bool stopOnFailMergeCom
             var latestChangesetId = GetLatestChangesetId();
             if (lastChangesetIdToFetch != -1)
                 latestChangesetId = Math.Min(latestChangesetId, lastChangesetIdToFetch);
-            // TFS 2010 doesn't like when we ask for history past its last changeset.
+
             if (MaxChangesetId >= latestChangesetId)
+            {
+                Trace.WriteLine("TFS 2010 doesn't like when we ask for history past its last changeset.");
```

Comment:
Can you change the message to something like "Skipping fetch, we already have X, but server only has Y"?